lib/resque/worker.rb

```diff
@@ -491,20 +491,18 @@ def self.all_heartbeats
     # Returns a list of workers that have sent a heartbeat in the past, but which
     # already expired (does NOT include workers that have never sent a heartbeat at all).
     def self.all_workers_with_expired_heartbeats
-      workers = Worker.all
       heartbeats = Worker.all_heartbeats
       now = data_store.server_time
 
-      workers.select do |worker|
-        id = worker.to_s
-        heartbeat = heartbeats[id]
-
+      heartbeats.select {|id, heartbeat|
         if heartbeat
           seconds_since_heartbeat = (now - Time.parse(heartbeat)).to_i
           seconds_since_heartbeat > Resque.prune_interval
         else
           false
         end
+      }.each_key.map do |id|
```

Comment:
`each_key` is generally designed to take a block, I think a better pattern here would be `keys.map do...`

Reply:
In this case, `keys` produces unnecessary array object and it's slightly slower than `each_key`.

```ruby
require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'

  gem 'benchmark-ips'
end

hash = 10000.times.map {|i| [i.to_s, i] }.to_h

Benchmark.ips do |x|
  x.report("each_key.map") do
    hash.each_key.map(&:to_i)
  end

  x.report("keys.map") do
    hash.keys.map(&:to_i)
  end

  x.compare!
end

__END__
$ ruby foo.rb
Warming up --------------------------------------
        each_key.map    79.000  i/100ms
            keys.map    97.000  i/100ms
Calculating -------------------------------------
        each_key.map    835.474  (± 1.6%) i/s -      4.187k in   5.012814s
            keys.map    971.279  (± 3.0%) i/s -      4.947k in   5.097821s

Comparison:
            keys.map:      971.3 i/s
        each_key.map:      835.5 i/s - 1.16x  (± 0.00) slower
```

https://github.com/JuanitoFatas/fast-ruby/tree/38f49f95fc7574d929de60b71791d09129c2588c#hasheach_key-instead-of-hashkeyseach-code

Is it ok?